apps/api-express/src/app/daos/search.ts

```diff
@@ -3,18 +3,63 @@ import {
   ParsedArticle,
   ParsedArticleParagraph,
   ScholarsDB,
+  ParsedArticlesByArticleId,
 } from '@foodmedicine/interfaces';
 import { runScholarsScraper } from '@foodmedicine/scholars-scraper';
 import * as articleParser from '@foodmedicine/article-parser';
-import { cleanString } from '@foodmedicine/word-explorer'
+import { cleanString } from '@foodmedicine/word-explorer';
+
+function groupParagraphsByArticle(
+  articlesStandalone: ParsedArticleParagraphStandalone[]
+): ParsedArticlesByArticleId[] {
+  const byIds: { [key: string]: ParsedArticleParagraphStandalone[] } = {};
+  articlesStandalone.map((articleStandalone) => {
+    if (!byIds[articleStandalone.head.id]) {
+      byIds[articleStandalone.head.id] = [];
+    }
+    byIds[articleStandalone.head.id].push(articleStandalone);
+  });
+  const parsedByTitle: ParsedArticlesByArticleId[] = Object.keys(byIds)
+    .map((id) => {
+      if (!byIds[id] || byIds[id].length === 0) {
```

Comment:
Minor nit: TS allows you to use optional chaining to make this a little more compact:

```
if (!byIds[id]?.length) {
    return null;
}